pubsub/api/Pubsub.Samples/PublishOrderedMessagesAsync.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START pubsub_quickstart_ordered_publisher]
+// [START pubsub_publish]
+
+using Google.Cloud.PubSub.V1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+public class PublishOrderedMessagesAsyncSample
+{
+    public async Task<int> PublishOrderedMessagesAsync(string projectId, string topicId, IEnumerable<KeyValuePair<string, string>> messagesAndKeys)
+    {
+        TopicName topicName = TopicName.FromProjectTopic(projectId, topicId);
+
+        var customSettings = new PublisherClient.Settings
+        {
+            EnableMessageOrdering = true
+        };
+
+        PublisherClient publisher = await PublisherClient.CreateAsync(topicName, settings: customSettings);
+
+        int publishedMessageCount = 0;
+        var publishTasks = messagesAndKeys.Select(async messageAndKey =>
+        {
+            try
+            {
+                string message = await publisher.PublishAsync(messageAndKey.Value, messageAndKey.Key);
```

Comment:
I believe this is swapped, the first parameter is the ordering key, and the sedcond is the actual message.
```suggestion
                string message = await publisher.PublishAsync(messageAndKey.Key, messageAndKey.Value);
```

Reply:
These were actually in the right order. The Value was the ordering key and the Key was the message because multiple messages shared the same key. I admit, this was confusing, so I moved to a List of Tuples with the key as the first element. Hopefully that is more clear.